meadow-client/src/MonadApp.purs

```diff
@@ -147,6 +134,30 @@ instance monadAppHalogenApp ::
   patchGistByGistId newGist gistId = runAjax $ Server.patchGistsByGistId newGist gistId
   postContractHaskell source = runAjax $ Server.postContractHaskell source
 
+-- I don't quite understand why but if you try to use MonadApp methods in HalogenApp methods you
+-- blow the stack so we have 3 methods pulled out here. I think this just ensures they are run
+-- in the HalogenApp monad and that's all that's required although a type annotation inside the
+-- monad doesn't seem to help, neither does `wrap . runHalogenApp`
+saveInitialState' :: forall m. MonadEffect m => HalogenApp m Unit
+saveInitialState' = do
+  oldContract <- marloweEditorGetValue'
+  modifying _oldContract
```

Comment:
FWIW I _think_ you can replace this with something like `modifying (_oldContract <<< at Nothing) (fromMaybe "" oldContract)` if you want to.

(That's the rough shape of it at least. Type checker's opinion is final.)

Reply:
doesn't quite work out, I'm sure there's something better here, will investigate at a later date.